### DISCUSSION TITLE
Road map for QIR support

### DISCUSSION THREAD

**Discussion by hamidelmaazouz** [CQCL/pytket-extensions#507] [General]
Hello @Roland-djee @cqc-alec,

I would like to know what the plans are for supporting more advanced QIR releases.

The QIR extension hosted at https://github.com/CQCL/pytket-qir supports only pyqir <0.7.0. Any plans for releases >= 0.7.0 and >= 0.8.0 ? If yes is somebody already working on it ? if yes when do you guys plan to release it ?

I would be happy to contribute  but don't want to start sth that is already in play.

Best,

**Comment by cqc-alec** [comment]
We definitely do plan to fully support QIR in the future, both as an input and as an output format, but we are still in the process of designing how this support will integrate into future TKET. Yes, pytket-qir was one experiment in that direction. Thank you for the offer to contribute: we will keep you posted as the design materializes. Meanwhile if you just want to apply pytket passes to subsections of QIR code (not crossing classical control boundaries), you could take a look at [pytket-qirpass](https://github.com/CQCL/pytket-qirpass).

**Reply by hamidelmaazouz** [reply]
Glad to hear the update.

Is there an ETA for an early a beta version ?

> Meanwhile if you just want to apply pytket passes to subsections of QIR code (not crossing classical control boundaries), you could take a look at [pytket-qirpass](https://github.com/CQCL/pytket-qirpass).

I am interested the passes but also in parsing and routing. I need to target QIR as high as 0.8.0. And I don't want to invest an effort (you seem to have already started) when a release would be expected soon.

Best,

**Reply by cqc-melf** [reply]
Hi @hamidelmaazouz 
just to give a short update: we have completed the update to pyqir 0.8.2 and there is now a release candidate that you could try to use if you want to. For more details please take a look on https://github.com/CQCL/pytket-qir
